### PR TITLE
add 1-27 as supported release branch

### PR DESCRIPTION
**Pull request by ahreehong**
*Issue #, if available:*

*Description of changes:*
Add EKS-D 1.27 build support
Add 1-27 as a supported release branch
Add 1-27 metrics server
Add 1-27 cluster autoscaler

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: UPSTREAM_PROJECTS.yaml [hunk 1]
```diff
@@ -133,6 +133,7 @@ projects:
           - tag: v1.24.1
           - tag: v1.25.0
           - tag: v1.26.0
+          - tag: v1.26.0
```
Comment (ahreehong) [inline UPSTREAM_PROJECTS.yaml:136]: Duplicated during generation bc we are temporarily using this tag for 1.26 and 1.27

File: projects/kubernetes-sigs/image-builder/Makefile [hunk 2]
```diff
@@ -49,6 +49,7 @@ ADDITIONAL_PAUSE_1-23_FROM=1-22
 ADDITIONAL_PAUSE_1-24_FROM=1-23
 ADDITIONAL_PAUSE_1-25_FROM=1-24
 ADDITIONAL_PAUSE_1-26_FROM=1-26
```
Comment (mitalipaygude) [inline projects/kubernetes-sigs/image-builder/Makefile:51]:
```suggestion
ADDITIONAL_PAUSE_1-26_FROM=1-25
```

shouldn't this be updated now?

File: UPSTREAM_PROJECTS.yaml [hunk 3]
```diff
@@ -121,6 +121,7 @@ projects:
           - commit: f69e14b5d
           - commit: e8d3e9b1d
           - commit: f48095c20
+          - commit: 299c9637229fb2bf849c1d86243fe2948d14101e
```
Comment (abhay-krishna) [inline UPSTREAM_PROJECTS.yaml:124]: nit: let's change the other commit references to full SHA hashes as well

File: projects/kubernetes/autoscaler/1-21/GIT_TAG
Comment (abhinavmpandey08) [inline projects/kubernetes/autoscaler/1-21/GIT_TAG:1]: We don't support 1.21 anymore. @abhay-krishna should we remove this?

Comment (abhinavmpandey08) [reply projects/kubernetes/autoscaler/1-21/GIT_TAG:1]: yeah let's track that in a separate ticket. Doesn't make sense to keep building it if it's not going to be used

**Pull request merged**